src/common/extensionMessaging.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as http from "http";
+
+/**
+ * Message server parameters.
+ */
+export let ServerParams = {
+    PORT: 8099,
+    HOST: "127.0.0.1"
+};
+
+/**
+ * Defines the messages sent to the extension.
+ * Add new messages to this enum.
+ */
+export enum ExtensionIncomingMessage {
+    START_PACKAGER,
+    STOP_PACKAGER
+}
+
+/**
+ * Generic interface for messages with arguments.
+ */
+export interface MessageWithArgs<T> {
```

Comment:
I'd put the classes and stuff shared between the client and the server in a common third file like ExtensionCommunicationProtocol.ts